src/index.js

```diff
@@ -22,25 +22,39 @@ const getOutput = (changes, domains) => {
 
 (async () => {
     const {pull_request} = github.context.payload;
-
-    if (!pull_request) {
-        core.error('Only pull requests events can trigger this action');
-    }
-
     const domainsString = core.getInput('domains', {required: true});
     const domains = JSON.parse(domainsString);
 
     const token = core.getInput('token', {required: true});
     const octokit = github.getOctokit(token);
 
-    const response = await octokit.paginate(octokit.rest.pulls.listFiles.endpoint.merge({
-        ...github.context.repo,
-        pull_number: pull_request.number,
-    }));
+    let files = [];
 
-    const output = getOutput(response.map(({filename}) => filename), domains);
+    switch (github.context.eventName) {
+        case 'pull_request': {
+            files = await octokit.paginate(octokit.rest.pulls.listFiles.endpoint.merge({
+                ...github.context.repo,
+                pull_number: pull_request.number,
```

Comment:
`pull_request` variable is being used only once and only when event is `pull_request` so we shall move declaration to inside this case